### PR TITLE
Docs: Add documentation for dashboard and folder roles

### PR DESCRIPTION
**Pull request by kalleep**
**What this PR does / why we need it**:
Document new fixed roles for dashboards and folders 

**Which issue(s) this PR fixes**:
Fixes #https://github.com/grafana/grafana-enterprise/issues/2585

### CODE REVIEW

File: docs/sources/enterprise/access-control/fine-grained-access-control-references.md [hunk 1]
```diff
@@ -38,14 +38,24 @@ The reference information that follows complements conceptual information about
...
+| `fixed:folders:reader`                 | `folders:read`<br>`dashboards:read`                                                                                                                                                                                                                                      | Read all folders and dashboards.                                                                                                                                                                                                                                                      |
+| `fixed:folders:writer`                 | All permissions from `fixed:dashboards:writer` and <br>`folders:read`<br>`folders:write`<br>`folders:create`<br>`folders:delete`<br>`folders.permissios:read`<br>`folders.permissions:write`                                                                             | Read, create, update and delete all folders and dashboards.                                                                                                                                                                                                                           |
```
Comment (achatterjee-grafana) [inline docs/sources/enterprise/access-control/fine-grained-access-control-references.md:50]:
```suggestion
| `fixed:folders:writer`                 | All permissions from `fixed:dashboards:writer` and <br>`folders:read`<br>`folders:write`<br>`folders:create`<br>`folders:delete`<br>`folders.permissios:read`<br>`folders.permissions:write`                                                                             | Read, create, update, and delete all folders and dashboards.                                                                                                                                                                                                                           |
```

File: docs/sources/enterprise/access-control/fine-grained-access-control-references.md [hunk 2]
```diff
@@ -38,14 +38,24 @@ The reference information that follows complements conceptual information about
...
+| `fixed:dashboards:reader`              | `dashboards:read`                                                                                                                                                                                                                                                        | Read all dashboards.                                                                                                                                                                                                                                                                  |
+| `fixed:dashboards:writer`              | All permissions from `fixed:dashboards:reader` and <br>`dashboards:write`<br>`dashboards:edit`<br>`dashboards:delete`<br>`dashboards:create`<br>`dashboards.permissios:read`<br>`dashboards.permissions:write`                                                           | Read, create, update and delete all dashboards.                                                                                                                                                                                                                                       |
```
Comment (achatterjee-grafana) [inline docs/sources/enterprise/access-control/fine-grained-access-control-references.md:45]:
```suggestion
| `fixed:dashboards:writer`              | All permissions from `fixed:dashboards:reader` and <br>`dashboards:write`<br>`dashboards:edit`<br>`dashboards:delete`<br>`dashboards:create`<br>`dashboards.permissios:read`<br>`dashboards.permissions:write`                                                           | Read, create, update, and delete all dashboards.                                                                                                                                                                                                                                       |
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added copy-edits.

File: docs/sources/enterprise/access-control/permissions.md [hunk 3]
```diff
@@ -118,3 +131,5 @@ The following list contains fine-grained access control scopes.
...
 | `provisioners:*`                                                                     | Restrict an action to a set of provisioners. For example, `provisioners:*` matches any provisioner, and `provisioners:accesscontrol` matches the fine-grained access control [provisioner]({{< relref "./provisioning.md" >}}).  |
 | `datasources:*`<br>`datasources:id:*`<br>`datasources:uid:*`<br>`datasources:name:*` | Restrict an action to a set of data sources. For example, `datasources:*` matches any data source, and `datasources:name:postgres` matches the data source named `postgres`.                                                     |
+| `folders:*`<br>`folders:id:*`                                                        | Restrict an action to a set of folders. For example, `folders:*` matches any folder, and `folders:id:1` matches the folder whose ID is `1`.                                                                                      |
```
Comment (IevaVasiljeva) [inline docs/sources/enterprise/access-control/permissions.md:134]: nit: maybe worth mentioning that this scope can also be applied to dashboards. Something like `Restrict an action to a set of folders or dashboards within those folders.`

**Review by IevaVasiljeva — APPROVED** [review]
Overall looks good, but I've added comments for some typos and some suggestions.

**Comment by kalleep** [comment]
> Needs backport label to show content in `latest`.

This is will be out in 8.5 so no backport needed @achatterjee-grafana

**Review by achatterjee-grafana — APPROVED** [review]

**Pull request merged**